doc/src/manual/calling-c-and-fortran-code.md

```diff
@@ -425,6 +425,35 @@ checks and is only meant to improve readability of the call.
     ccall(:main, Int32, (Int32, Ptr{Ptr{UInt8}}), length(argv), argv)
     ```
 
+!!! note
+    For Fortran functions taking variable length strings of type `character(len=*)` the string lengths
+    are provided as *hidden arguments*. Type and position of these arguments in the list are compiler
+    specific, where compiler vendors usually default to using `Csize_t` as type and append the hidden
+    arguments at the end of the argument list. While this behaviour is fixed for some compilers (GNU),
+    others *optionally* permit placing hidden arguments directly after the character argument (Intel,PGI). 
+    For example, Fortran subroutines of the form
+
+    ```fortran
+    subroutine test(str1, str2)
+    character(len=*) :: str1,str2
+    ```
+
+    can be called via the following Julia code, where the lengths are appended
+
+    ```julia
+    str1 = "example1"
+    str2 = "example2"
+    ccall(:test, Void, (Ptr{UInt8}, Ptr{UInt8}, Csize_t, Csize_t), 
+                        str1, str2, length(str1), length(str2))
```

Comment:
Should probably be `sizeof` for number of bytes. `length` is the number of unicode codepoints.

Reply:
Since Fortran and C don't speak unicode `sizeof` and `length` should ideally be identical. Can change it though.

Reply:
Well `length` is O(N) and it's the wrong query. In practice it probably doesn't matter since at worst you'll get a short string, but it's the wrong predicate for a byte string.

Reply:
Fixed it.